modules/xmpp/xmpp.js

```diff
@@ -508,8 +514,9 @@ export default class XMPP extends Listenable {
         this._maybeSendDeploymentInfoStat(true);
 
         if (this._components.length > 0) {
```

Comment:
Drop the if since it's always initialized to an array?

Reply:
By default components is empty and we will iterate over the messages and don't do anything. So its not a big deal with the check ...

Reply:
I understand that functionally it wont change anything. I'm asking technically what is the advantage to iterate over the messages and call the function so that it will quickly exit versus just having the check?